app/controllers/plugins_controller.rb

```diff
@@ -40,6 +40,7 @@ def update
     if find_plugin_setting
       @plugin_setting.disabled = value_to_boolean(params[:plugin_setting][:disabled]) if params[:plugin_setting] && !params[:plugin_setting][:disabled].nil?
       @plugin_setting.posted_settings = params[:settings] || {} unless @plugin_setting.disabled
+      @plugin_setting.settings[:secret] ||= params[:settings][:secret] if @plugin_setting.name == "big_blue_button" && @plugin_setting.settings
```

Comment:
You should not be special casing individual plugins in this controller. You should be modifying the validator for that plugin instead